cpp/arcticdb/storage/azure/azure_storage.cpp

```diff
@@ -47,6 +47,61 @@ namespace fg = folly::gen;
 
 namespace detail {
 
+// TODO: fix this temporary workaround to read error code. azure-sdk-cpp client sometimes doesn't properly set the error code.
+//  This issue has been raised on the sdk repo. Once fixed, we should no longer need the following function and would just read e.ErrorCode.
+std::string get_error_code(const Azure::Core::RequestFailedException& e) {
+    auto error_code = e.ErrorCode;
+
+    if(error_code.empty() && e.RawResponse ) {
+        auto headers_map = e.RawResponse->GetHeaders();
+        if(headers_map.find("x-ms-error-code") != headers_map.end()) {
+            error_code = headers_map["x-ms-error-code"];
+        }
+    }
+    return error_code;
+}
+
+void raise_azure_exception(const Azure::Core::RequestFailedException& e) {
+    auto error_code = get_error_code(e);
+    auto status_code = e.StatusCode;
+    std::string error_message;
+
+    if(status_code == Azure::Core::Http::HttpStatusCode::NotFound && error_code == AzureErrorCode_to_string(AzureErrorCode::BlobNotFound)) {
```

Comment:
Maybe we can reform the check then raise logic to `storage::check<ErrorCode::E_DUPLICATE_KEY>(...` to align with the practices we have elsewhere?

Reply:
I think you mean `storage::check<ErrorCode::E_KEY_NOT_FOUND>(...`? In other storages, this is always thrown as `KeyNotFoundException` as there is a specific class defined for it. I think it is better to leave it as it is for consistency